### PR TITLE
starship: new, 1.0.0

### PR DESCRIPTION
**Pull request by CoelacanthusHex**
Topic Description
-----------------

Introduce `starship`, a customizable prompt for shell written by Rust.

Package(s) Affected
-------------------

`starship`

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Comment by CoelacanthusHex** [comment]
LGTM on AMD64, bash, zsh and fish work well
图片

File: extra-utils/starship/autobuild/beyond [hunk 1]
```diff
@@ -0,0 +1,10 @@
+declare -A completion_files=([bash]="$PKGDIR/usr/share/bash-completion/completions/starship"
+                             [zsh]="$PKGDIR/usr/share/zsh/site-functions/_starship"
+                             [fish]="$PKGDIR/usr/share/fish/vendor_completions.d/starship.fish"
+                             )
+
+for sh in 'bash' 'fish' 'zsh'; do
+    abinfo "Generating completion file for $sh ..."
+    mkdir -p $(dirname ${completion_files[$sh]})
```
Comment (WhiredPlanck) [inline extra-utils/starship/autobuild/beyond]: Add verbosity here please.

File: extra-utils/starship/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,9 @@
+PKGNAME=starship
+PKGSEC=utils
+PKGDES="A customizable prompt for shell."
```
Comment (WhiredPlanck) [inline extra-utils/starship/autobuild/defines]: It would be better to remove the period.

File: extra-utils/starship/autobuild/beyond (hunk 1 above)
Comment (liushuyu) [inline extra-utils/starship/autobuild/beyond]:
```suggestion
    mkdir -pv "$(dirname ${completion_files[$sh]})"
```

Prevent potential globbing and add verbosity.

File: extra-utils/starship/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+PKGDEP="glibc openssl libgit2"
+PKGSUG="bash zsh fish powershell tcsh elvish"
+BUILDDEP="rustc llvm python-3"
+ABTYPE=rust
+USECLANG=1
+ABSPLITDBG=0
```
Comment (liushuyu) [inline extra-utils/starship/autobuild/defines:10]:
```suggestion

ABTYPE=rust
USECLANG=1
ABSPLITDBG=0
```

Usually we use a blank line to separate metadata definitions and compilation definitions.

**Review by WhiredPlanck — CHANGES_REQUESTED** [review]

File: extra-utils/starship/autobuild/defines (hunk 2 above)
Comment (liushuyu) [inline extra-utils/starship/autobuild/defines]:
```suggestion
PKGDES="A customizable shell prompt"
```

Remove period. Also, rephrase the sentence if needed, you don't have to follow what the upstream says.

**Review by MingcongBai — APPROVED** [review]

**Review by WhiredPlanck — APPROVED** [review]

**Pull request merged**